File: day01/src/main.py
#!/usr/bin/env python

with open("./data/input.txt") as input:
    lines = input.readlines()

depths = list(map(lambda x: int(x), lines))
couples = zip(depths[:-1], depths[1:])
count = len([x for (x, y) in couples if x < y])
print("count:", count)

depths3 = list(map(sum, zip(depths[:-2], depths[1:-1], depths[2:])))
couples = zip(depths3[:-1], depths3[1:])
count = len([x for (x, y) in couples if x < y])
print("count:", count)

File: day01/src/main.rs
use std::fs;

fn main() {

    let input = fs::read_to_string("./data/input.txt")
        .expect("Something went wrong reading the file");

    let depths: Vec<usize> = input
        .lines()
        .map(|s| s.parse::<usize>().unwrap())
        .collect();

    let counter = depths
        .iter()
        .zip(depths.iter().skip(1))
        .filter(|(a , b)| a < b)
        .count();

    println!("Counter: {:?}", counter);

    let depths3: Vec<usize> = depths
        .windows(3)
        .map(|win| win.iter().sum::<usize>())
        .collect();

    let counter3 = depths3
        .iter()
        .zip(depths3.iter().skip(1))
        .filter(|(a, b)| a < b).
        count();

    println!("Counter3: {:?}", counter3);

}
